### PR TITLE
21 deadlock integration

### LINKED ISSUE #21: Deadlock Integration
image

### LINKED ISSUE #23: urgent add of /hayes command
image
image
image
image

### PR DESCRIPTION
**Pull request by whig123**
3 commands added
/deadlockpatchnotes
/deadlockstats
/deadlocklinkaccount
/hayes

### CODE REVIEW

File: commands/hayes.js
Comment (Nuozrok) [inline commands/hayes.js:1]: what on earth

File: data/accountscache.json
Comment (Nuozrok) [inline data/accountscache.json:1]: Is this safe to keep on the GitHub? I guess these are all public numbers/ids?

File: data/exampledeadlockdata.json
Comment (Nuozrok) [inline data/exampledeadlockdata.json:1]: Example file. Redundant?

File: data/exampledeadlockpatchnotes.json
Comment (Nuozrok) [inline data/exampledeadlockpatchnotes.json:1]: Example file. Redundant?

File: data/hayesdata.txt
Comment (Nuozrok) [inline data/hayesdata.txt:1]: Weird formatting in this. Just look at the first line. 'IPAcenrVrfrdaudioenusRutherfordoga'

File: package.json
Comment (Nuozrok) [inline package.json:1]: Will this break distube as described in issue #22?

File: data/accountscache.json
Comment (Nuozrok) [reply data/accountscache.json:1]: GitGuardian check deems it safe.

**Review by Nuozrok — CHANGES_REQUESTED** [review]

File: commands/hayes.js
Comment (whig123) [reply commands/hayes.js:1]: looks good to me

File: data/accountscache.json
Comment (whig123) [reply data/accountscache.json:1]: this is all public info. the steamID is what you see on their profile. the rest of it is all details you can see on someone else's profile

File: data/exampledeadlockdata.json
Comment (whig123) [reply data/exampledeadlockdata.json:1]: yeah we don't need it, deleted

File: data/hayesdata.txt
Comment (whig123) [reply data/hayesdata.txt:1]: the people get what they want, they asked for hayes, I deliver hayes

File: package.json
Comment (whig123) [reply package.json:1]: what version of node is production bot running on? if it is running on 18.20.5 LTS or earlier we should be fine, if it is running later then yes but the bot would already be broken

**Review by whig123 — COMMENTED** [review]
give the deadlock players what they deserve

File: package.json
Comment (Nuozrok) [reply package.json:1]: Screenshot 2024-12-19 at 6 41 41 PM

Comment (Nuozrok) [reply package.json:1]: should be good then

Comment (whig123) [reply package.json:1]: let me test on my machine though while we still have 2 copies

File: data/experimental/deadlocksingularmatchexample.json
Comment (Nuozrok) [inline data/experimental/deadlocksingularmatchexample.json:1]: example? redundant?

File: package.json
Comment (whig123) [reply package.json:1]: to test the /play command

File: data/experimental/example_protobuf.bz2
Comment (Nuozrok) [inline data/experimental/example_protobuf.bz2:1]: example? redundant?

File: data/experimental/example_protobuf/example_protobuf
Comment (Nuozrok) [inline data/experimental/example_protobuf/example_protobuf:1]: example? redundant?

File: data/exampledeadlockdata.json
Comment (Nuozrok) [inline data/exampledeadlockdata.json:1]: delete empty file

**Review by Nuozrok — CHANGES_REQUESTED** [review]
address comments

File: package.json
Comment (whig123) [reply package.json:1]: did you have to set up ffmpeg originally to get distube to work? it is asking me to install that on my pc when I'm testing that distube still works

Comment (Nuozrok) [reply package.json:1]: yes

**Review by whig123 — COMMENTED** [review]
give the people what they want2

**Comment by Nuozrok** [comment]
@whig123  please document commands on the [wiki](https://github.com/Nuozrok/lpap/wiki/Commands)

**Comment by whig123** [comment]
give the people what they want3

**Review by Nuozrok — APPROVED** [review]

**Pull request merged**